RFCs/0000-expand-new-task.md

```diff
@@ -0,0 +1,158 @@
+
+### Add `--private` and `--remote` flags to `pdk new task`
+
+- When specified, these flags will simply pre-populate the generated Bolt Task metadata with `"private": true` and `"remote": true` key/value pairs respectively.
+
+### Add optional `<implementations>` argument to `pdk new task`
+
+- Allow `pdk new task` to accept an unbounded, space-separated list of `implementations` to be provided after the task name.
+
+- When one or more implementation files are provided, PDK will only generate a `tasks/<taskname>.json` file for the new task. The generated metadata will contain an `"implementations"` key with each given implementation file listed.
+
+- If the `--language` option is provided at the same time as one or more `<implementation>` arguments, PDK should exit with an appropriate error message.
+
+#### Examples
+
+The following example illustrates how a user might use PDK to author a new "cross-platform" task with Windows- and Linux-specific implementations:
+
+```bash
+$ pdk new task --private --language=bash sql_linux
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_linux.sh' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_linux.json' from template.
+
+$ pdk new task --private --language=powershell sql_windows
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_windows.ps1' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_windows.json' from template.
+
+$ pdk new task sql tasks/sql_linux.sh tasks/sql_windows.ps1
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json' from template.
+pdk (INFO): Added implementation 'sql_linux.sh' to '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json'
+pdk (INFO): Added implementation 'sql_windows.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json'
+```
+
+Given the following task implementation script already exists at `tasks/init.rb`:
+
+```ruby
+#!/usr/bin/env ruby
+require 'json'
+
+params = JSON.parse(STDIN.read)
+action = params['action'] || params['_task']
+if ['start',  'stop'].include?(action)
+  `systemctl #{params['_task']} #{params['service']}`
+end
+```
+
+The following example illustrates how a user might use PDK to author a new set of "shared-implementation" tasks
+
+```bash
+$ pdk new task start tasks/init.rb
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/start.json' from template.
+pdk (INFO): Added implementation 'init.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/start.json'
+
+$ pdk new task stop tasks/init.rb
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/stop.json' from template.
+pdk (INFO): Added implementation 'init.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/stop.json'
+```
+
+## Unresolved Questions
+
+- Should `--private` and/or `--remote` be disallowed in combination with `<implementations>`?
```

Comment:
It seems like "no"? Unless there is some specific reason that a "cross-platform" or "shared implementation" task couldn't also be `private` or `remote`?

Reply:
I don't think there is any specific relationship that would cause it to be a bad idea to use them together. \ however `remote` may be ambiguous as the key can live at the top level or inside an implementation